site/docs/uk/hosting/fly.md

```diff
@@ -0,0 +1,437 @@
+```
+
+</CodeGroupItem>
+</CodeGroup>
+
+3. **Deno**: змініть версію Deno та видаліть `CMD`, якщо він існує, у файлі `Dockerfile`.
+   Наприклад, у цьому випадку ми оновлюємо `DENO_VERSION` до `1.25.2`.
+
+   **Node.js**: щоб змінити версію Node.js, вам потрібно вставити властивість `"node"` в межах властивості `"engines"` у `package.json`.
+   Ми оновлюємо версію Node.js до `16.14.0` у наступному прикладі.
+
+<CodeGroup>
+<CodeGroupItem title="Deno" Active>
+
+```dockerfile{2,26}
+# Dockerfile
+ARG DENO_VERSION=1.25.2
+ARG BIN_IMAGE=denoland/deno:bin-${DENO_VERSION}
+FROM ${BIN_IMAGE} AS bin
+
+FROM frolvlad/alpine-glibc:alpine-3.13
+
+RUN apk --no-cache add ca-certificates
+
+RUN addgroup --gid 1000 deno \
+  && adduser --uid 1000 --disabled-password deno --ingroup deno \
+  && mkdir /deno-dir/ \
+  && chown deno:deno /deno-dir/
+
+ENV DENO_DIR /deno-dir/
+ENV DENO_INSTALL_ROOT /usr/local
+
+ARG DENO_VERSION
+ENV DENO_VERSION=${DENO_VERSION}
+COPY --from=bin /deno /bin/deno
+
+WORKDIR /deno-dir
+COPY . .
+
+ENTRYPOINT ["/bin/deno"]
+# CMD видалено
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="Node.js" Active>
+
+```json{19}
+// package.json
+{
+  "name": "grammy",
+  "version": "1.0.0",
+  "description": "grammy",
+  "main": "app.js",
+  "author": "itsmeMario",
+  "license": "MIT",
+  "dependencies": {
+    "express": "^4.18.1",
+    "grammy": "^1.11.0"
+  },
+  "devDependencies": {
+    "@types/express": "^4.17.14",
+    "@types/node": "^18.7.18",
+    "typescript": "^4.8.3"
+  },
+  "engines": {
+    "node": "16.14.0"
+  }
+}
+```
+
+</CodeGroupItem>
+</CodeGroup>
+
+4. Відредагуйте `app` у файлі `fly.toml`.
+   Шлях `./app.ts` або `./app.js` для Node.js у наведеному нижче прикладі вказує на розташування головного файлу.
+   Ви можете змінити їх, щоб вони відповідали каталогу вашого проєкту.
+   Якщо ви використовуєте вебхуки, переконайтеся, що порт відповідає тому, що ви вказали у своєму [конфігураційному файлі](#вебхуки) (`8000`).
+
+<CodeGroup>
+<CodeGroupItem title="Deno (Вебхук)" Active>
+
+```toml{7,11,12}
+# fly.toml
+app = "grammy"
+kill_signal = "SIGINT"
+kill_timeout = 5
+
+[processes]
+  app = "run --allow-net ./app.ts"
+
+[[services]]
+  http_checks = []
+  internal_port = 8000
+  processes = ["app"]
+  protocol = "tcp"
+  script_checks = []
+  [services.concurrency]
+    hard_limit = 25
+    soft_limit = 20
+    type = "connections"
+
+  [[services.ports]]
+    force_https = true
+    handlers = ["http"]
+    port = 80
+
+  [[services.ports]]
+    handlers = ["tls", "http"]
+    port = 443
+
+  [[services.tcp_checks]]
+    grace_period = "1s"
+    interval = "15s"
+    restart_limit = 0
+    timeout = "2s"
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="Deno (Тривале опитування)" Active>
+
+```toml{7}
+# fly.toml
+app = "grammy"
+kill_signal = "SIGINT"
+kill_timeout = 5
+
+[processes]
+  app = "run --allow-net ./app.ts"
+
+# Simply omitting the whole [[services]] section
+# since we are not listening to HTTP
```

Comment:
I think this part is safe for translation.

Reply:
```suggestion
# Просто опускаємо весь розділ [[services]],
# оскільки ми не прослуховуємо HTTP
```